specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json

```diff
@@ -777,6 +828,272 @@
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SearchServiceNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The operation succeeded. The response contains the list of all private endpoint connections for the given service.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnectionListResult"
+            }
+          },
+          "default": {
+            "description": "An unexpected error occurred during the operation.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Search/searchServices/{searchServiceName}/sharedPrivateLinkResources/{sharedPrivateLinkResourceName}": {
+      "put": {
+        "tags": [
+          "SharedPrivateLinkResources"
+        ],
+        "operationId": "SharedPrivateLinkResources_CreateOrUpdate",
+        "x-ms-examples": {
+          "SharedPrivateLinkResourceCreateOrUpdate": {
+            "$ref": "./examples/CreateOrUpdateSharedPrivateLinkResource.json"
+          }
+        },
+        "description": "Creates or updates a shared private link resource managed by the Search service in the given resource group.",
+        "externalDocs": {
+          "url": "https://aka.ms/search-manage"
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SearchServiceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SharedPrivateLinkResourceNameParameter"
+          },
+          {
+            "name": "sharedPrivateLinkResource",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SharedPrivateLinkResource"
+            },
+            "description": "The definition of the shared private link resource to create or update."
+          },
+          {
+            "$ref": "#/parameters/ClientRequestIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The existing shared private link resource definition was successfully updated.",
+            "schema": {
+              "$ref": "#/definitions/SharedPrivateLinkResource"
+            }
+          },
+          "201": {
+            "description": "The shared private link resource was successfully created.",
+            "schema": {
+              "$ref": "#/definitions/SharedPrivateLinkResource"
+            }
```

Comment:
is this async ? 

        "x-ms-long-running-operation": true,
        "x-ms-long-running-operation-options": {
          "final-state-via": "azure-async-operation"  or "location" based on how async is communicated
        }

Reply:
It is not async so this is unnecessary.